sys-libs/libcxx/libcxx-9999.ebuild

```diff
@@ -9,13 +9,13 @@ EGIT_REPO_URI="http://llvm.org/git/libcxx.git
 	https://github.com/llvm-mirror/libcxx.git"
 CMAKE_MIN_VERSION=3.4.3
 
-[ "${PV%9999}" != "${PV}" ] && SCM="git-r3" || SCM=""
+[[ ${PV} == 9999 ]] && SCM="git-r3" || SCM=""
```

Comment:
It's not a regexp but a pattern, which is simpler than a regexp and the idiomatic way of doing prefix/suffix matches in shell.

Yes, your method is certainly not ugly. It only does:
1. copy a string,
2. check if the string ends with a fixed substring (which is also processed as a pattern),
3. strip the suffix if that is the case,
4. compare the resulting string with the original.
This is certainly the most efficient and readable way of doing it.

Next thing I know, the best way to get all `.txt` files from a directory is to:

    while read -r -d '' f
    do
        if [[ ${f%.txt} != ${f} ]]; then
            :
        fi
    done < <(find . -maxdepth 1 -type f -print0)